vpr/src/place/move_transactions.h

```diff
@@ -48,17 +48,23 @@ struct t_pl_blocks_to_be_moved {
     t_pl_blocks_to_be_moved(const t_pl_blocks_to_be_moved&) = delete;
     t_pl_blocks_to_be_moved(t_pl_blocks_to_be_moved&&) = delete;
 
-/**
- * @brief This function increments the size of the moved_blocks vector and return the index
- * of the newly added last elements. 
- */
+    /**
+    * @brief This function increments the size of the moved_blocks vector and return the index
+    * of the newly added last elements.
+    */
     size_t get_size_and_increment();
 
-/**
- * @brief This function clears all data structures of this struct.
- */
+    /**
+    * @brief This function clears all data structures of this struct.
+    */
     void clear_move_blocks();
 
+    /**
+     * @param net The unique identifier of the net of interest.
+     * @return True if the driver block of the net is among the moving blocks.
+     */
+    bool driven_by_moved_block(const ClusterNetId net) const;
+
```

Comment:
These three functions are `BlkLocRegistry` modifiers. Is moving them to `BlkLocRegistry` class a good idea?
```
apply_move_blocks()
commit_move_blocks()
revert_move_blocks()

```